### PR TITLE
Add docs for Datadog APM with Istio service mesh

### PR DESCRIPTION
**Pull request by cgilmour**
### What does this PR do?
Adds docs for users to configure their Istio service mesh and collect traces with Datadog.

### Motivation
The feature is available and user-facing documentation is missing.

### Preview link
https://docs-staging.datadoghq.com/cgilmour/istio-docs/tracing/setup/istio/

### CODE REVIEW

File: content/en/tracing/setup/istio.md [hunk 1]
```diff
@@ -14,29 +14,27 @@ further_reading:
...
 
 ## Configuration
 
 ### Datadog Agent Installation
 
-The Datadog Agent must be [installed][k8s-install] and [APM enabled][apm-enabled].
-The `hostPort` setting needs to be uncommented for Istio sidecars to connect to the agent and submit traces.
+1. [Install the Agent][1]
+2. [Make sure APM is enabled for your Agent][2].
+3. Set the `apm_non_local_traffic` parameter to `true` and uncomment the `hostPort` setting so that Istio sidecars can connect to the Agent and submit traces.
```
Comment (cgilmour) [inline content/en/tracing/setup/istio.md]: This gets automatically enabled when `KUBERNETES=yes` and `DD_APM_ENABLED=true`.
It's not an additional step a user needs to do.

Comment (l0k0ms) [reply content/en/tracing/setup/istio.md]: Updated :) does the new wording works for you?

Comment (cgilmour) [reply content/en/tracing/setup/istio.md]: Yup!

**Comment by cgilmour** [comment]
Removed WIP / Do Not Merge labels.
Could we get a code-owner review? Thanks!

**Pull request merged**